internal/relationship/by_file_ownership.go

```diff
@@ -135,6 +162,43 @@ func findOwnershipByFilesRelationships(resolver file.Resolver, catalog *pkg.Coll
 	return relationships
 }
 
+func directOwnersByPath(catalog *pkg.Collection) map[string]map[string][]artifact.ID {
+	directOwnership := make(map[string]map[string][]artifact.ID)
```

Comment:
What would you think about simplifying this to: `map[string][]pkg.Type`? I think it could make both the population and usage above simpler, I think.

Reply:
Answer from team discussion: it seems more-or-less the safest to check within the same type, as implemented here 👍

Reply:
Okay cool! Sorry for the delay, been at an in-person summit thing this week. Yeah I could go either way too, but it seemed more semantically correct in my head (for hand-waivy reasons) for the interpretation of symlinks for a given installation DB to be scoped to that same DB

Reply:
@luhring do you mind if I push a couple tweaks to this for performance/clarity?

Reply:
Not at all! Thanks! 🙏